CMakeLists.txt

```diff
@@ -1,281 +1,294 @@
 
-if (EVHTP_HAS_STACK_PROTECTOR)
-    set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fstack-protector-strong")
+if(NOT EVHTP_DISABLE_EVTHR)
+    find_package(Threads REQUIRED)
+    if(CMAKE_THREAD_LIBS_INIT)
+        list(APPEND LIBEVHTP_SOURCE_FILES  thread.c)
+        list(APPEND LIBEVHTP_EXTERNAL_LIBS Threads::Threads)
+        list(APPEND package_deps Threads)
+        list(APPEND public_headers "include/evhtp/thread.h")
+    endif()
 endif()
 
-if (EVHTP_HAS_C99)
-    add_definitions(-DEVHTP_HAS_C99)
+# TODO: Add Oniguruma to Hunter
+# if(NOT EVHTP_DISABLE_REGEX)
+#     find_package(Oniguruma CONFIG REQUIRED)
+#     list(APPEND LIBEVHTP_EXTERNAL_LIBS ${ONIGURUMA_LIBRARIES})
+#     list(APPEND package_deps Oniguruma)
+# endif()
+
+# TODO: Add jemalloc and tcmalloc to Hunter
+# if(EVHTP_ALLOCATOR STREQUAL "jemalloc")
+#     find_package(jemalloc CONFIG REQUIRED)
+#     list(APPEND LIBEVHTP_EXTERNAL_LIBS jemalloc)
+#     list(APPEND package_deps jemalloc)
+# elseif(EVHTP_ALLOCATOR STREQUAL "tcmalloc")
+#     find_package(tcmalloc CONFIG REQUIRED)
+#     list(APPEND LIBEVHTP_EXTERNAL_LIBS tcmalloc)
+#     list(APPEND package_deps tcmalloc)
+# endif()
+set(EVHTP_DISABLE_REGEX ON)
+
+add_library(evhtp ${LIBEVHTP_SOURCE_FILES})
+target_link_libraries(evhtp PUBLIC ${LIBEVHTP_EXTERNAL_LIBS})
+
+target_compile_definitions(evhtp PUBLIC "-DPROJECT_VERSION=${PROJECT_VERSION}")
+if(has_stack_protector)
+    target_compile_options(evhtp PUBLIC -fstack-protector-strong)
 endif()
 
-if (EVHTP_HAS_VISIBILITY_HIDDEN)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden")
+if(EVHTP_THR_SHARED_PIPE)
+    target_compile_definitions(evhtp PUBLIC -DEVTHR_SHARED_PIPE)
 endif()
 
-if (NOT HAVE_SYS_TREE)
-    configure_file (
-        ${PROJECT_SOURCE_DIR}/compat/sys/tree.h.in
-        ${PROJECT_BINARY_DIR}/compat/sys/tree.h)
+if(has_c99)
+    target_compile_definitions(evhtp PUBLIC -DEVHTP_HAS_C99)
+endif()
 
-    include_directories(${PROJECT_BINARY_DIR}/compat)
+if(NOT HAVE_SYS_TREE)
+    configure_file(
+        ${CMAKE_CURRENT_SOURCE_DIR}/compat/sys/tree.h.in
+        ${CMAKE_CURRENT_BINARY_DIR}/compat/sys/tree.h)
+    list(APPEND compat_headers "${CMAKE_CURRENT_BINARY_DIR}/compat/sys/tree.h")
 endif()
 
-if (NOT HAVE_SYS_QUEUE)
-    configure_file (
-        ${PROJECT_SOURCE_DIR}/compat/sys/queue.h.in
-        ${PROJECT_BINARY_DIR}/compat/sys/queue.h)
+if(NOT HAVE_SYS_QUEUE)
+    configure_file(
+        ${CMAKE_CURRENT_SOURCE_DIR}/compat/sys/queue.h.in
+        ${CMAKE_CURRENT_BINARY_DIR}/compat/sys/queue.h)
+    list(APPEND compat_headers "${CMAKE_CURRENT_BINARY_DIR}/compat/sys/queue.h")
+endif()
 
-    include_directories(${PROJECT_BINARY_DIR}/compat)
+if(NOT HAVE_SYS_TREE OR NOT HAVE_SYS_QUEUE)
+    target_include_directories(evhtp PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/compat>)
 endif()
 
-if (NOT HAVE_STRNDUP)
-    add_definitions(-DNO_STRNDUP)
+if(NOT HAVE_STRNDUP)
+    target_compile_definitions(evhtp PUBLIC -DNO_STRNDUP)
```

Comment:
Is it really working? I excepting `target_compile_definitions(evhtp PUBLIC NO_STRNDUP)` (without `-D`).

Reply:
I guess I have `strndup` so I wouldn't know if this actually works. I'll fix it.